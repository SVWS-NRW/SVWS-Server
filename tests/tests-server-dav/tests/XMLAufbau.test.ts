import { describe, expect, test } from "vitest";
import { getApiService } from "./utils/RequestBuilder.js"

describe("Adressbuch Dav", () => {
	const apiService = getApiService('Admin', '')

	// Verschiedene Bodies die bei Webdav anfragen verwendet werden
	const bodyDataDav: string = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
			"<D:propfind\n" +
			"\txmlns:D='DAV:'\n" +
			"\txmlns:A='http://apple.com/ns/ical/'\n" +
			"\txmlns:C='urn:ietf:params:xml:ns:caldav'>\n" +
			"\t<D:prop>\n" +
			"\t\t<D:resourcetype/>\n" +
			"\t\t<D:owner/>\n" +
			"\t\t<D:displayname/>\n" +
			"\t\t<D:current-user-principal/>\n" +
			"\t\t<D:current-user-privilege-set/>\n" +
			"\t\t<A:calendar-color/>\n" +
			"\t\t<C:calendar-home-set/>\n" +
			"\t</D:prop>\n" +
			"</D:propfind>\n"

	const bodyDataAdressbuecher: string = "<propfind xmlns=\"DAV:\"\n" +
			"\t\t  xmlns:card='urn:ietf:params:xml:ns:carddav'>\n" +
			"\t<prop>\n" +
			"\t\t<resourcetype/>\n" +
			"\t\t<displayname/>\n" +
			"\t\t<card:supported-address-data/>\n" +
			"\t</prop>\n" +
			"</propfind>"

	const bodyDataAdressbuchSchueler: string = "<propfind xmlns=\"DAV:\"\n" +
			"\txmlns:card=\"urn:ietf:params:xml:ns:carddav\"\n" +
			"\txmlns:cs=\"http://calendarserver.org/ns/\" xmlns:d=\"DAV:\">\n" +
			"\t<prop>\n" +
			"\t\t<resourcetype />\n" +
			"\t\t<getetag />\n" +
			"\t\t<cs:getctag />\n" +
			"\t</prop>\n" +
			"</propfind>\n"

	const bodyreportDataAdressbuchSchueler: string = "<card:addressbook-multiget\n" +
			"\txmlns:card=\"urn:ietf:params:xml:ns:carddav\"\n" +
			"\txmlns:cs=\"http://calendarserver.org/ns/\"\n" +
			"\txmlns:d=\"DAV:\">\n" +
			"\t<d:prop>\n" +
			"\t\t<d:getetag/>\n" +
			"\t\t<card:address-data/>\n" +
			"\t</d:prop>\n" +
			"\t<d:href>/dav/gymabi/adressbuecher/schueler/Schueler_1001.vcf</d:href>\n" +
			"\t<d:href>/dav/gymabi/adressbuecher/schueler/Schueler_1002.vcf</d:href>\n" +
			"</card:addressbook-multiget>\n"

	const bodySyncCollection: string = "<sync-collection xmlns=\"DAV:\" xmlns:card=\"urn:ietf:params:xml:ns:carddav\" xmlns:cs=\"http://calendarserver.org/ns/\" xmlns:d=\"DAV:\">\n" +
			"<sync-token>0</sync-token>\n" +
			"<sync-level>1</sync-level>\n" +
			"<prop>\n" +
			"<getetag/>\n" +
			"<card:address-data/>\n" +
			"</prop>\n" +
			"</sync-collection>"

	const bodyAdressbuecherErzieherIssue1283 = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
			"<C:addressbook-multiget xmlns:D=\"DAV:\"\n" +
			"\txmlns:C=\"urn:ietf:params:xml:ns:carddav\">\n" +
			"\t<D:prop>\n" +
			"\t\t<D:getetag />\n" +
			"\t\t<C:address-data Content-Type='text/vcard' version='3.0' />\n" +
			"\t</D:prop>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2095.vcf</D:href>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2098.vcf</D:href>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2105.vcf</D:href>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2107.vcf</D:href>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2114.vcf</D:href>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2288.vcf</D:href>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2289.vcf</D:href>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2292.vcf</D:href>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2481.vcf</D:href>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2294.vcf</D:href>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2295.vcf</D:href>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2298.vcf</D:href>\n" +
			"\t<D:href>/dav/gymabi/adressbuecher/erzieher/Erzieher_2307.vcf</D:href>\n" +
			"</C:addressbook-multiget>\n"

	describe.each([{schema: "GymAbiDav"}])('Adressbücher Dav Tests gegen %s', ({schema}) => {
		test("Prüft das XML Format gleich geblieben ist.", async () => {
			const response = await apiService.propfind(`/dav/${schema}`, {body: bodyDataDav})
			const xmlAsString = await response!.text()

			expect(response).toBeDefined();
			expect(response!.status).toBe(207);
			expect(xmlAsString).toMatchSnapshot()
		});

		test("Prüft das XML Format für Adressbücher gleich geblieben ist.", async () => {
			const response = await apiService.propfind(`/dav/${schema}/adressbuecher`, {body: bodyDataAdressbuecher})
			const xmlAsString = await response!.text()

			expect(response).toBeDefined();
			expect(response!.status).toBe(207);
			expect(xmlAsString).toMatchSnapshot()
		});

		// Derzeit deaktiviert da sehr großes Output in den Snapshots
		test.skip("Prüft das XML Format für Adressbücher der Schüler gleich geblieben ist.", async () => {
			const response = await apiService.propfind(`/dav/${schema}/adressbuecher/schueler`, {body: bodyDataAdressbuchSchueler})
			const xmlAsString = await response!.text()

			expect(response).toBeDefined();
			expect(response!.status).toBe(207);
			expect(xmlAsString).toMatchSnapshot()
		});

		test("Testet Report-Anfrage auf notwendige Informationen.", async () => {
			const response = await apiService.report(`/dav/${schema}/adressbuecher/schueler`, {body: bodyreportDataAdressbuchSchueler})
			const xmlAsString = await response!.text()

			expect(response).toBeDefined();
			expect(response!.status).toBe(207);
			expect(xmlAsString).toMatchSnapshot()
		});

		// Derzeit deaktiviert da sehr großes Output in den Snapshots
		test.skip("Prüft das XML Format für Sync-Collection-Anfrage.", async () => {
			const response = await apiService.report(`/dav/${schema}/adressbuecher/schueler`, {body: bodySyncCollection})
			const xmlAsString = await response!.text()

			expect(response).toBeDefined();
			expect(response!.status).toBe(207);
			expect(xmlAsString).toMatchSnapshot()
		});

		/**
		 * Test zum Issue 1283 - Unmarshalling funktionierte nicht, wenn das
		 * ContentType-Attribut im CardAddressData großgeschrieben wurde.
		 */
		// Dieser Test schlägt fehl
		test.skip("Issue 1283 - Unmarshalling funktionierte nicht", async () => {
			const response = await apiService.propfind(`/dav/${schema}/adressbuecher/schueler`, {body: bodyAdressbuecherErzieherIssue1283})
			const xmlAsString = await response!.text()

			expect(response).toBeDefined();
			expect(response!.status).toBe(207);
			expect(xmlAsString).toMatchSnapshot()
		});
	});

	describe("Kalender Dav", () => {
		const apiServiceAnde = getApiService('Ande', '')
		const apiServiceBagi = getApiService('BAGI', '')

		const bodyDataDavCalCollection: string = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
				"<D:propfind xmlns:D='DAV:'\n" +
				"\txmlns:A='http://apple.com/ns/ical/'>\n" +
				"\t<D:prop>\n" +
				"\t\t<D:resourcetype />\n" +
				"\t\t<D:displayname />\n" +
				"\t\t<D:current-user-privilege-set />\n" +
				"\t\t<A:calendar-color />\n" +
				"\t</D:prop>\n" +
				"</D:propfind>\n"

		const bodyDataDavCal: string = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
				"<D:propfind\n" +
				"\txmlns:D='DAV:'\n" +
				"\txmlns:C='urn:ietf:params:xml:ns:caldav'\n" +
				"\txmlns:CS='http://calendarserver.org/ns/'>\n" +
				"\t<D:prop>\n" +
				"\t\t<D:resourcetype/>\n" +
				"\t\t<D:owner/>\n" +
				"\t\t<D:current-user-principal/>\n" +
				"\t\t<D:current-user-privilege-set/>\n" +
				"\t\t<D:supported-report-set/>\n" +
				"\t\t<C:supported-calendar-component-set/>\n" +
				"\t\t<CS:getctag/>\n" +
				"\t</D:prop>\n" +
				"</D:propfind>\n"

		describe.each([{schema: "GymAbiDav"}])('Kalender Dav Tests gegen %s', ({schema}) => {
			test("Testet Property-Suche auf die Liste von  mit Account Ande", async () => {
				const response = await apiServiceAnde.propfind(`/dav/${schema}/kalender`, {body: bodyDataDavCalCollection})
				const xmlAsString = await response!.text()

				expect(response).toBeDefined();
				expect(response!.status).toBe(207);
				expect(xmlAsString).toMatchSnapshot()
			});

			test("Testet Property-Suche auf die Liste von Kalendern mit Account Bagi", async () => {
				const response = await apiServiceBagi.propfind(`/dav/${schema}/kalender`, {body: bodyDataDavCalCollection})
				const xmlAsString = await response!.text()

				expect(response).toBeDefined();
				expect(response!.status).toBe(207);
				expect(xmlAsString).toMatchSnapshot()
			});

			test("Ruft ein Propfind auf den Gemeinsamen Kalender auf.", async () => {
				const response = await apiServiceAnde.propfind(`/dav/${schema}/kalender/1`, {body: bodyDataDavCal})
				const xmlAsString = await response!.text()

				expect(response).toBeDefined();
				expect(response!.status).toBe(207);
				expect(xmlAsString).toMatchSnapshot()
			});
		});
	});
});
