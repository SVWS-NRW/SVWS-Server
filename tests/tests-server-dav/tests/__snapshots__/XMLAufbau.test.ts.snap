// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Adressbuch Dav > Adressbücher Dav Tests gegen { schema: 'GymAbiDav' } > Prüft das XML Format für Adressbücher gleich geblieben ist. 1`] = `"<?xml version="1.0" encoding="UTF-8" standalone="yes"?><d:multistatus xmlns:ical="http://apple.com/ns/ical" xmlns:cal="urn:ietf:params:xml:ns:caldav" xmlns:card="urn:ietf:params:xml:ns:carddav" xmlns:cs="http://calendarserver.org/ns/" xmlns:d="DAV:"><d:response><d:href>/dav/GymAbiDav/adressbuecher/schueler</d:href><d:propstat><d:prop><d:displayname>Schüler</d:displayname><d:resourcetype><d:collection/><card:addressbook/></d:resourcetype><card:supported-address-data><card:addressDataTypes content-type="text/vcard" version="4.0"/></card:supported-address-data></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/dav/GymAbiDav/adressbuecher/lehrer</d:href><d:propstat><d:prop><d:displayname>Lehrer</d:displayname><d:resourcetype><d:collection/><card:addressbook/></d:resourcetype><card:supported-address-data><card:addressDataTypes content-type="text/vcard" version="4.0"/></card:supported-address-data></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/dav/GymAbiDav/adressbuecher/erzieher</d:href><d:propstat><d:prop><d:displayname>Erzieher</d:displayname><d:resourcetype><d:collection/><card:addressbook/></d:resourcetype><card:supported-address-data><card:addressDataTypes content-type="text/vcard" version="4.0"/></card:supported-address-data></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response></d:multistatus>"`;

exports[`Adressbuch Dav > Adressbücher Dav Tests gegen { schema: 'GymAbiDav' } > Prüft das XML Format für Adressbücher gleich geblieben ist. 2`] = `"<?xml version="1.0" encoding="UTF-8" standalone="yes"?><d:multistatus xmlns:ical="http://apple.com/ns/ical" xmlns:cal="urn:ietf:params:xml:ns:caldav" xmlns:card="urn:ietf:params:xml:ns:carddav" xmlns:cs="http://calendarserver.org/ns/" xmlns:d="DAV:"><d:response><d:href>/dav/GymAbiDav/adressbuecher/schueler</d:href><d:propstat><d:prop><d:displayname>Schüler</d:displayname><d:resourcetype><d:collection/><card:addressbook/></d:resourcetype><card:supported-address-data><card:addressDataTypes content-type="text/vcard" version="4.0"/></card:supported-address-data></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/dav/GymAbiDav/adressbuecher/lehrer</d:href><d:propstat><d:prop><d:displayname>Lehrer</d:displayname><d:resourcetype><d:collection/><card:addressbook/></d:resourcetype><card:supported-address-data><card:addressDataTypes content-type="text/vcard" version="4.0"/></card:supported-address-data></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response><d:response><d:href>/dav/GymAbiDav/adressbuecher/erzieher</d:href><d:propstat><d:prop><d:displayname>Erzieher</d:displayname><d:resourcetype><d:collection/><card:addressbook/></d:resourcetype><card:supported-address-data><card:addressDataTypes content-type="text/vcard" version="4.0"/></card:supported-address-data></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response></d:multistatus>"`;

exports[`Adressbuch Dav > Adressbücher Dav Tests gegen { schema: 'GymAbiDav' } > Prüft das XML Format gleich geblieben ist. 1`] = `"<?xml version="1.0" encoding="UTF-8" standalone="yes"?><d:multistatus xmlns:ical="http://apple.com/ns/ical" xmlns:cal="urn:ietf:params:xml:ns:caldav" xmlns:card="urn:ietf:params:xml:ns:carddav" xmlns:cs="http://calendarserver.org/ns/" xmlns:d="DAV:"><d:response><d:href>/dav/GymAbiDav</d:href><d:propstat><d:prop><d:resourcetype><d:collection/><d:principal/></d:resourcetype><d:current-user-principal><d:href>/dav/GymAbiDav/benutzer/1</d:href></d:current-user-principal><d:current-user-privilege-set><d:privilege><d:read-current-user-privilege-set/></d:privilege><d:privilege><d:read/></d:privilege><d:privilege><d:read-acl/></d:privilege></d:current-user-privilege-set><cal:calendar-home-set><d:href>/dav/GymAbiDav/kalender</d:href></cal:calendar-home-set></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><d:owner/><d:displayname/><ical:calendar-color/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response><d:response><d:href>/dav/GymAbiDav/adressbuecher</d:href><d:propstat><d:prop><d:resourcetype><d:collection/></d:resourcetype><d:current-user-principal><d:href>/dav/GymAbiDav/benutzer/1</d:href></d:current-user-principal><d:current-user-privilege-set><d:privilege><d:read-current-user-privilege-set/></d:privilege><d:privilege><d:read/></d:privilege><d:privilege><d:read-acl/></d:privilege></d:current-user-privilege-set></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><d:owner/><d:displayname/><cal:calendar-home-set/><ical:calendar-color/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response><d:response><d:href>/dav/GymAbiDav/kalender</d:href><d:propstat><d:prop><d:resourcetype><d:collection/></d:resourcetype><d:current-user-principal><d:href>/dav/GymAbiDav/benutzer/1</d:href></d:current-user-principal><d:current-user-privilege-set><d:privilege><d:read-current-user-privilege-set/></d:privilege><d:privilege><d:read/></d:privilege><d:privilege><d:read-acl/></d:privilege></d:current-user-privilege-set><cal:calendar-home-set><d:href>/dav/GymAbiDav/kalender</d:href></cal:calendar-home-set></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><d:owner/><d:displayname/><ical:calendar-color/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response></d:multistatus>"`;

exports[`Adressbuch Dav > Adressbücher Dav Tests gegen { schema: 'GymAbiDav' } > Testet Report-Anfrage auf notwendige Informationen. 1`] = `"<?xml version="1.0" encoding="UTF-8" standalone="yes"?><d:multistatus xmlns:ical="http://apple.com/ns/ical" xmlns:cal="urn:ietf:params:xml:ns:caldav" xmlns:card="urn:ietf:params:xml:ns:carddav" xmlns:cs="http://calendarserver.org/ns/" xmlns:d="DAV:"/>"`;

exports[`Adressbuch Dav > Kalender Dav > Kalender Dav Tests gegen { schema: 'GymAbiDav' } > Ruft ein Propfind auf den Gemeinsamen Kalender auf. 1`] = `"<?xml version="1.0" encoding="UTF-8" standalone="yes"?><d:multistatus xmlns:ical="http://apple.com/ns/ical" xmlns:cal="urn:ietf:params:xml:ns:caldav" xmlns:card="urn:ietf:params:xml:ns:carddav" xmlns:cs="http://calendarserver.org/ns/" xmlns:d="DAV:"><d:response><d:href>/dav/GymAbiDav/kalender/1</d:href><d:propstat><d:prop><d:owner><d:href>/dav/GymAbiDav/benutzer/5</d:href></d:owner><d:resourcetype><d:collection/><cal:calendar/></d:resourcetype><d:supported-report-set><d:supported-report><d:report><cal:calendar-multiget/></d:report></d:supported-report><d:supported-report><d:report><cal:calendar-query/></d:report></d:supported-report><d:supported-report><d:report><d:sync-collection/></d:report></d:supported-report></d:supported-report-set><d:current-user-principal><d:href>/dav/GymAbiDav/benutzer/5</d:href></d:current-user-principal><d:current-user-privilege-set><d:privilege><d:read-current-user-privilege-set/></d:privilege><d:privilege><d:read/></d:privilege><d:privilege><d:read-acl/></d:privilege><d:privilege><d:all/></d:privilege><d:privilege><d:write/></d:privilege></d:current-user-privilege-set><cs:getctag>1737973043000</cs:getctag><cal:supported-calendar-component-set><cal:calendarComponents name="VEVENT"/></cal:supported-calendar-component-set></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response></d:multistatus>"`;

exports[`Adressbuch Dav > Kalender Dav > Kalender Dav Tests gegen { schema: 'GymAbiDav' } > Testet Property-Suche auf die Liste von  mit Account Ande 1`] = `"<?xml version="1.0" encoding="UTF-8" standalone="yes"?><d:multistatus xmlns:ical="http://apple.com/ns/ical" xmlns:cal="urn:ietf:params:xml:ns:caldav" xmlns:card="urn:ietf:params:xml:ns:carddav" xmlns:cs="http://calendarserver.org/ns/" xmlns:d="DAV:"><d:response><d:href>/dav/GymAbiDav/kalender</d:href><d:propstat><d:prop><d:resourcetype><d:collection/></d:resourcetype><d:current-user-privilege-set><d:privilege><d:read-current-user-privilege-set/></d:privilege><d:privilege><d:read/></d:privilege><d:privilege><d:read-acl/></d:privilege></d:current-user-privilege-set></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><d:displayname/><ical:calendar-color/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response><d:response><d:href>/dav/GymAbiDav/kalender/1</d:href><d:propstat><d:prop><d:displayname>Gemeinsamer Kalender</d:displayname><d:resourcetype><d:collection/><cal:calendar/></d:resourcetype><d:current-user-privilege-set><d:privilege><d:read-current-user-privilege-set/></d:privilege><d:privilege><d:read/></d:privilege><d:privilege><d:read-acl/></d:privilege><d:privilege><d:all/></d:privilege><d:privilege><d:write/></d:privilege></d:current-user-privilege-set></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><ical:calendar-color/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response><d:response><d:href>/dav/GymAbiDav/kalender/3</d:href><d:propstat><d:prop><d:displayname>Eigener Kalender</d:displayname><d:resourcetype><d:collection/><cal:calendar/></d:resourcetype><d:current-user-privilege-set><d:privilege><d:read-current-user-privilege-set/></d:privilege><d:privilege><d:read/></d:privilege><d:privilege><d:read-acl/></d:privilege><d:privilege><d:all/></d:privilege><d:privilege><d:write/></d:privilege></d:current-user-privilege-set></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><ical:calendar-color/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response></d:multistatus>"`;

exports[`Adressbuch Dav > Kalender Dav > Kalender Dav Tests gegen { schema: 'GymAbiDav' } > Testet Property-Suche auf die Liste von Kalendern mit Account Bagi 1`] = `"<?xml version="1.0" encoding="UTF-8" standalone="yes"?><d:multistatus xmlns:ical="http://apple.com/ns/ical" xmlns:cal="urn:ietf:params:xml:ns:caldav" xmlns:card="urn:ietf:params:xml:ns:carddav" xmlns:cs="http://calendarserver.org/ns/" xmlns:d="DAV:"><d:response><d:href>/dav/GymAbiDav/kalender</d:href><d:propstat><d:prop><d:resourcetype><d:collection/></d:resourcetype><d:current-user-privilege-set><d:privilege><d:read-current-user-privilege-set/></d:privilege><d:privilege><d:read/></d:privilege><d:privilege><d:read-acl/></d:privilege></d:current-user-privilege-set></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><d:displayname/><ical:calendar-color/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response><d:response><d:href>/dav/GymAbiDav/kalender/1</d:href><d:propstat><d:prop><d:displayname>Gemeinsamer Kalender</d:displayname><d:resourcetype><d:collection/><cal:calendar/></d:resourcetype><d:current-user-privilege-set><d:privilege><d:read-current-user-privilege-set/></d:privilege><d:privilege><d:read/></d:privilege><d:privilege><d:read-acl/></d:privilege></d:current-user-privilege-set></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><ical:calendar-color/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response><d:response><d:href>/dav/GymAbiDav/kalender/4</d:href><d:propstat><d:prop><d:displayname>Eigener Kalender</d:displayname><d:resourcetype><d:collection/><cal:calendar/></d:resourcetype><d:current-user-privilege-set><d:privilege><d:read-current-user-privilege-set/></d:privilege><d:privilege><d:read/></d:privilege><d:privilege><d:read-acl/></d:privilege><d:privilege><d:all/></d:privilege><d:privilege><d:write/></d:privilege></d:current-user-privilege-set></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><ical:calendar-color/></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response></d:multistatus>"`;
